ui/common/src/timing.ts

```diff
@@ -168,3 +168,32 @@ export function debounce<T extends (...args: any) => void>(
       }, wait);
   };
 }
+
+export function browserTaskQueueMonitor(interval = 1000): { wasSuspended: boolean; reset: () => void } {
```

Comment:
this is the logic for determining a task queue suspension occurred. false positives are possible in extreme scheduling scenarios, but in the socket case they just result in a rare misclassification of event types in the gathered data.